src/Server.php

```diff
@@ -111,6 +112,11 @@ public function authenticate(string $username, string $password): ConnectionInte
         }
 
         $check = \imap_check($resource);
+
+        if (false === $check) {
+            throw new ResourceCheckFailureException('Resource check failure');
```

Comment:
I suggest instead of the message "Resource check failure" add the result of the output function imap_last_error() otherwise it is not clear because of what the error.

Reply:
All our exceptions do this, check out https://github.com/ddeboer/imap/blob/master/src/Exception/AbstractException.php